### PR TITLE
Updatetomongodb

### PR DESCRIPTION
**Pull request by SvLowman**

### CODE REVIEW

File: client/src/pages/GalleryPage.js [hunk 1]
```diff
@@ -10,31 +10,27 @@ const Display = styled.div`
...
         </section>
         <Display>
-          <img src="" alt="testImage" />
           <p>Display für Cloudinary-Links von MongoDB</p>
           {loading && <p>Loading...</p>}
           {error && <p>{error.message}</p>}
-          {console.log(images)}
-          {data &&
-            images.map((image) => (
+          {console.log(userData)}
```
Comment (domko95) [inline client/src/pages/GalleryPage.js:31]: console.logs are great for development, but should not end up in production.

File: client/src/pages/TaggingPage.js [hunk 2]
```diff
@@ -1,35 +1,60 @@
...
+const TaggingPage = () => {
+  const url =
+    "https://res.cloudinary.com/tagtrace/image/upload/v1606822076/TagTrace/wmcwfmhypvc06pnjtcpg.jpg";
+  const imgNr = "001";
+  const userName = "sven";
+  const [tagName, setTagName] = useState("");
+
+  const handleTagNameChange = (event) => {
+    const tagName = event.target.value;
+    console.log("tagName:", tagName);
```
Comment (domko95) [inline client/src/pages/TaggingPage.js:22]: see above 😃

File: client/src/pages/TaggingPage.js [hunk 3]
```diff
@@ -1,35 +1,60 @@
...
+    setTagName(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!tagName) {
+      console.log("No tag name");
+    } else {
+      console.log("The fabulous tagName is:", tagName);
```
Comment (domko95) [inline client/src/pages/TaggingPage.js:31]: Think about switching from console.logs to something that your users can see.

File: client/src/utils/api.js [hunk 4]
```diff
@@ -1,5 +1,33 @@
 export const getImageObj = async (userName) => {
   const response = await fetch(`/api/users/${userName}`);
   const imgEntry = await response.json();
+  console.log(imgEntry);
```
Comment (domko95) [inline client/src/utils/api.js:4]: see above

File: client/src/utils/api.js [hunk 5]
```diff
@@ -1,5 +1,33 @@
...
+    method: "POST",
+    body: JSON.stringify({ tagName }),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  console.log("tagName as provided in api.js:", tagName);
+  console.log("Response:", response);
+  const newTagArray = await response.json();
+  console.log(("newTagArray:", newTagArray));
```
Comment (domko95) [inline client/src/utils/api.js:31]: see above

File: client/src/utils/api.js [hunk 6]
```diff
@@ -1,5 +1,33 @@
...
+
+export async function addNewTag(userName, imgNr, tagName) {
+  const response = await fetch(`/api/users/${userName}/images/${imgNr}/tags`, {
+    method: "POST",
+    body: JSON.stringify({ tagName }),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  console.log("tagName as provided in api.js:", tagName);
```
Comment (domko95) [inline client/src/utils/api.js:28]: see above

File: lib/data.js [hunk 7]
```diff
@@ -1,23 +1,54 @@
...
-    })
-    .toArray();
+async function getImageDataOfUser(userName) {
+  const imgEntry = await collection("users").findOne({
+    userName: userName,
+  });
   if (!imgEntry) {
     return null;
   }
   console.log({ imgEntry });
```
Comment (domko95) [inline lib/data.js:11]: see above

File: lib/data.js [hunk 8]
```diff
@@ -1,23 +1,54 @@
...
+    { $push: { images: imageObj } }
+  );
+}
 
-// async function getByID({ collectionName, id }) {
-//   const cursor = collection(collectionName).findOne({
-//     _id: ObjectId(id),
+// async function setImage(imageObj, userName) {
+//   const userData = await collection("users").findOne({
+//     name: userName,
```
Comment (domko95) [inline lib/data.js:35]: remove comments, if you don't need them.

File: lib/data.js [hunk 9]
```diff
@@ -1,23 +1,54 @@
...
 //   });
-//   return await cursor;
+//   const newImageArray = [...userData.images, imageObj];
+//   return await collection("users").updateOne(
+//     { name: userName },
+//     { $set: { images: newImageArray } }
+//   );
```
Comment (domko95) [inline lib/data.js:41]: remove comments, if you don't need them.

File: lib/data.js [hunk 10]
```diff
@@ -1,23 +1,54 @@
...
+  return await collection("users").updateOne(
+    { "images.imgNr": imgNr },
+    { $push: { "images.$.tags": tagName } }
+  );
+}
+
+exports.setImage = setImage;
+exports.setTag = setTag;
+exports.getImageDataOfUser = getImageDataOfUser;
+// exports.getImagesByTag = getImagesByTag;
```
Comment (domko95) [inline lib/data.js:54]: remove comments, if you don't need them

**Review by domko95 — COMMENTED** [review]
LGTM 👍🏻

File: client/src/pages/GalleryPage.js [hunk 11]
```diff
@@ -10,31 +10,27 @@ const Display = styled.div`
...
 const GalleryPage = () => {
   const userName = "sven";
-  //const tag = "2";
-  const { data, loading, error, doFetch } = useAsync(() =>
+  const { data: userData, loading, error, doFetch } = useAsync(() =>
     getImageObj(userName)
   );
 
   useEffect(() => {
     doFetch();
```
Comment (mhstrkmp) [inline client/src/pages/GalleryPage.js:18]: Not exactly sure why but your deployment fails and it seems to be a problem here ...

---
Error Message:
```
Failed to compile.

src/pages/GalleryPage.js
  Line 19:6:  React Hook useEffect has a missing dependency: 'doFetch'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
```

File: server.js [hunk 12]
```diff
@@ -1,49 +1,79 @@
 require("dotenv").config();
 
-const { cloudinary } = require("./lib/cloudinary");
 const express = require("express");
 const path = require("path");
 
-const { getImageDataByTag } = require("./lib/data");
+const { getImageDataOfUser } = require("./lib/data");
+const { setImage, setTag } = require("./lib/data");
```
Comment (mhstrkmp) [inline server.js:7]:
```suggestion
const { getImageDataOfUser, setImage, setTag } = require("./lib/data");
```

**Review by mhstrkmp — APPROVED** [review]
lgtm 🚀

File: client/src/pages/GalleryPage.js (hunk 1 above)
Comment (SvLowman) [reply client/src/pages/GalleryPage.js:31]: 👍 You're right of course!

File: client/src/pages/TaggingPage.js (hunk 3 above)
Comment (SvLowman) [reply client/src/pages/TaggingPage.js:31]: 👍

File: lib/data.js (hunk 8 above)
Comment (SvLowman) [reply lib/data.js:35]: 👍

File: server.js (hunk 12 above)
Comment (SvLowman) [reply server.js:7]: 👍

**Pull request merged**